### PR TITLE
feat: Add an empty user-agent header for the API calls

### LINKED ISSUE #1105: CLI tools - use empty user-agent
**Description**
We want to add an empty `user-agent` header for the API calls.

**Acceptance Criteria**
- [ ] an empty `user-agent` header is added for the API calls

### PR DESCRIPTION
**Pull request by solid-maxim**
Fixes #1105 

## Checklist

- [x] Documented
- [x] Unit tested

### CODE REVIEW

File: metrics/ci_integrations/lib/client/buildkite/buildkite_client.dart [hunk 1]
```diff
@@ -96,6 +96,7 @@ class BuildkiteClient with LoggerMixin {
     return <String, String>{
       HttpHeaders.contentTypeHeader: ContentType.json.value,
       HttpHeaders.acceptHeader: ContentType.json.value,
+      HttpHeaders.userAgentHeader: null,
```
Comment (solid-vladlaba) [inline metrics/ci_integrations/lib/client/buildkite/buildkite_client.dart:99]: What if we introduce the optional `headers` parameter for the client? There also be a private parameter `_headers` with values to apply to all API calls as follows:
```suggestion
      if (_headers != null) ..._headers,
```
In the constructor: `: _headers = headers;`
Thus, we can separate our requirements related to an empty `user-agent` from the low-level client implementation.

**Review by solid-vladlaba — CHANGES_REQUESTED** [review]
Well done, @solid-maxim @solid-tetianamovlian! I've added one suggestion below - please let me know what you think. Also, please apply this suggestion to all of the clients in this PR.

File: metrics/ci_integrations/lib/source/buildkite/client_factory/buildkite_source_client_factory.dart [hunk 2]
```diff
@@ -23,9 +25,13 @@ class BuildkiteSourceClientFactory
     ArgumentError.checkNotNull(config, 'config');
 
     final authorization = BearerAuthorization(config.accessToken);
+    final headers = {
+      HttpHeaders.userAgentHeader: null,
+    };
```
Comment (solid-vladlaba) [inline metrics/ci_integrations/lib/source/buildkite/client_factory/buildkite_source_client_factory.dart:30]: Would it be better to introduce some class with constants to place there this map with the default headers for the CI Integrations tool? So, here and in other places we can just `SomeConstants.defaultHeaders`. Also, can this map be `const`?
```suggestion
    const headers = {
      HttpHeaders.userAgentHeader: null,
    };
```

File: metrics/ci_integrations/test/client/buildkite/buildkite_client_test.dart [hunk 3]
```diff
@@ -141,6 +141,28 @@ void main() {
...
+          organizationSlug: organizationSlug,
+          authorization: authorization,
+          headers: additionalHeaders,
+        );
+
+        final headers = client.headers;
+
+        final expectedHeader = additionalHeaders.entries.first;
+
+        expect(headers, containsPair(expectedHeader.key, expectedHeader.value));
```
Comment (solid-vladlaba) [inline metrics/ci_integrations/test/client/buildkite/buildkite_client_test.dart:162]:
```suggestion
        const expectedKey = 'test-header';
        const expectedValue = 'test-value';
        const additionalHeaders = {
          expectedKey: expectedValue,
        };

        final client = BuildkiteClient(
          buildkiteApiUrl: buildkiteMockServer.url,
          organizationSlug: organizationSlug,
          authorization: authorization,
          headers: additionalHeaders,
        );
        final headers = client.headers;
        
        expect(headers, containsPair(expectedKey, expectedValue));
```
Please apply to similar test cases.

**Review by solid-vladlaba — DISMISSED** [review]
Looks better, @solid-maxim @solid-tetianamovlian! A few more small suggestions below - please let me know what you think!

**Review by illia-romanenko — APPROVED** [review]
LGTM - thanks @solid-maxim @solid-tetianamovlian !

**Pull request merged**